addons/binding/org.openhab.binding.dwdunwetter/src/main/java/org/openhab/binding/dwdunwetter/internal/DwdUnwetterHandler.java

```diff
@@ -0,0 +1,224 @@
+            }
+            if (data.isNew(i)) {
+                triggerChannel(getChannelUuid(CHANNEL_UPDATED, i), "NEW");
+            }
+        }
+        data.updateCache();
+    }
+
+    /**
+     * Returns the configuration.
+     *
+     * @return The configuration
+     */
+    private DwdUnwetterConfiguration getDwdConfig() {
+        if (config == null) {
+            config = getConfigAs(DwdUnwetterConfiguration.class);
+        }
+        return config;
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Start initializing!");
+        updateStatus(ThingStatus.UNKNOWN);
+
+        data = new DwdWarningsData(getDwdConfig().getCellId());
+
+        List<Channel> channels = createChannels();
+        updateThing(editThing().withChannels(channels).build());
+
+        startAutomaticRefresh();
+
+        updateStatus(ThingStatus.ONLINE);
+        logger.debug("Finished initializing!");
+    }
+
+    private ChannelUID getChannelUuid(String typeId, int warningNumber) {
+        return new ChannelUID(getThing().getUID(), typeId + (warningNumber + 1));
+    }
+
+    private ChannelUID getChannelUuid(String typeId) {
+        return new ChannelUID(getThing().getUID(), typeId);
+    }
+
+    /**
+     * Creates a trigger Channel.
+     */
+    private Channel createTriggerChannel(String typeId, String label, int warningNumber) {
+        ChannelUID channelUID = getChannelUuid(typeId, warningNumber);
+        return ChannelBuilder.create(channelUID, "String") //
+                .withType(new ChannelTypeUID(DwdUnwetterBindingConstants.BINDING_ID, typeId)) //
+                .withLabel(label + " (" + (warningNumber + 1) + ")")//
+                .withKind(ChannelKind.TRIGGER) //
+                .build();
+    }
+
+    /**
+     * Creates a normal, state based, channel associated with a warning.
+     */
+    private Channel createChannel(String typeId, String itemType, String label, int warningNumber) {
+        ChannelUID channelUID = getChannelUuid(typeId, warningNumber);
+        return ChannelBuilder.create(channelUID, itemType) //
+                .withType(new ChannelTypeUID(DwdUnwetterBindingConstants.BINDING_ID, typeId)) //
+                .withLabel(label + " (" + (warningNumber + 1) + ")")//
+                .build();
+    }
+
+    /**
+     * Creates a normal, state based, channel not associated with a warning.
+     */
+    private Channel createChannel(String typeId, String itemType, String label) {
+        ChannelUID channelUID = getChannelUuid(typeId);
+        return ChannelBuilder.create(channelUID, itemType) //
+                .withType(new ChannelTypeUID(DwdUnwetterBindingConstants.BINDING_ID, typeId)) //
+                .withLabel(label)//
+                .build();
+    }
+
+    /**
+     * Creates the Channel for each warning.
+     *
+     * @return The List of Channels
+     */
+    private List<Channel> createChannels() {
+        List<Channel> channels = new LinkedList<>();
```

Comment:
`ArrayList` would be a little more efficient.
(Could event be allocated with `warningCount  + 1`)

Reply:
11*warningCount + 1 would be the right number, but you are right.